### PR TITLE
Redesign

### PR DESCRIPTION
**Pull request by blink1073**
filebrowser_demo

### CODE REVIEW

**Comment by ellisonbg** [comment]
Looks like great progress!

**Comment by blink1073** [comment]
Now with last modified!

screen shot 2015-12-07 at 4 11 42 pm

**Comment by dwillmer** [comment]
That's awesome :)

**Comment by rgbkrk** [comment]
:+1:

**Comment by cameronoelsen** [comment]
:hand: High five! Looking great!

**Comment by jasongrout** [comment]
I want to cheer you on too!  Thanks!

**Comment by blink1073** [comment]
Cleaned up and updated the demo to include a `jupyter-js-editor` see original [comment](https://github.com/jupyter/jupyter-js-filebrowser/pull/6#issue-120868851).

**Comment by blink1073** [comment]
If no one has any objections, I'd like to merge this as a replacement for the existing functionality, and add more in further PRs.

**Comment by ellisonbg** [comment]
+1

On Tue, Dec 8, 2015 at 11:54 AM, Steven Silvester notifications@github.com
wrote:

> If no one has any objections, I'd like to merge this as a replacement for
> the existing functionality, and add more in further PRs.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jupyter/jupyter-js-filebrowser/pull/6#issuecomment-162996015
> .

## 

Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by rgbkrk** [comment]
I'd totally click Merge if the WIP tag was off. :wink:

**Comment by blink1073** [comment]
I refactored to get logic out of the view, WIP removed, ready for review.

**Comment by blink1073** [comment]
Switched back to WIP, refactoring.

**Comment by blink1073** [comment]
Done tweaking.

File: src/index.ts [hunk 1]
```diff
@@ -14,88 +33,216 @@ import './index.css';
...
+    }
+  }
+
+  /**
+   * Refresh the model contents.
+   */
+  refresh() {
+    this._contents.listContents(this._path).then(model => {
+      this._items = [];
+      // Add a link to the parent directory if not at the root.
```
Comment (blink1073) [inline src/index.ts:174]: I'll remove this once the breadcrumbs are done.

**Comment by dwillmer** [comment]
:+1: looks awesome, Steve!

**Pull request merged**